src/main/java/fr/minuskube/inv/content/InventoryContents.java

```diff
@@ -460,443 +499,519 @@ InventoryContents fillRect(int fromRow, int fromColumn,
+            Preconditions.checkArgument(pattern.isWrapAround(),
+                "To fill in a repeating pattern wrapAround needs to be enabled for the pattern to work!");
+            if (endRow < 0) {
+                endRow = this.inventory.getRows();
+            }
+            if (endColumn < 0) {
+                endColumn = this.inventory.getColumns();
+            }
             Preconditions.checkArgument(startRow < endRow, "The start row needs to be lower than the end row");
-            Preconditions.checkArgument(startColumn < endColumn, "The start column needs to be lower than the end column");
-
-            int rowDelta = endRow - startRow, columnDelta = endColumn - startColumn;
-            for(int row = 0; row <= rowDelta; row++) {
-                for(int column = 0; column <= columnDelta; column++) {
-                    ClickableItem item = pattern.getObject(row, column);
-
-                    if(item != null)
-                        set(startRow + row, startColumn + column, item);
+            Preconditions.checkArgument(startColumn < endColumn,
+                "The start column needs to be lower than the end column");
+            final int rowDelta = endRow - startRow;
+            final int columnDelta = endColumn - startColumn;
+            for (int row = 0; row <= rowDelta; row++) {
+                for (int column = 0; column <= columnDelta; column++) {
+                    final ClickableItem item = pattern.getObject(row, column);
+                    if (item != null) {
+                        this.set(startRow + row, startColumn + column, item);
+                    }
                 }
             }
             return this;
         }
 
+        @NotNull
         @Override
-        public InventoryContents fillPatternRepeating(Pattern<ClickableItem> pattern, SlotPos startPos, SlotPos endPos) {
-            return fillPatternRepeating(pattern, startPos.getRow(), startPos.getColumn(), endPos.getRow(), endPos.getColumn());
-        }
-
-        @Override
-        public InventoryContents fillPattern(Pattern<ClickableItem> pattern, int startRow, int startColumn) {
-            for(int row = 0; row < pattern.getRowCount(); row++) {
-                for(int column = 0; column < pattern.getColumnCount(); column++) {
-                    ClickableItem item = pattern.getObject(row, column);
-
-                    if(item != null)
-                        set(startRow + row, startColumn + column, item);
-                }
-            }
-
-            return this;
+        public InventoryContents fillPatternRepeating(@NotNull final Pattern<ClickableItem> pattern,
+                                                      @NotNull final SlotPos startPos, @NotNull final SlotPos endPos) {
+            return this.fillPatternRepeating(pattern, startPos.getRow(), startPos.getColumn(), endPos.getRow(),
+                endPos.getColumn());
         }
 
         @SuppressWarnings("unchecked")
+        @NotNull
```

Comment:
Actually this can be null if the map doesn't contain the element or the element is null. This is the first case where I noticed this in your annotations stuff. Perhaps the others have the same problem...